tests/nataliia_romanchuk/homework_4/test_collections.py

```diff
@@ -0,0 +1,424 @@
+    assert sorted_set == ['a', 'b', 'c', 'd']
+
+    a, b, c, d = some_dict
+    assert (a, b, c, d) == ('a', 'b', 'c', 'd')
+
+    sorted_set = sorted(some_set)
+    a, b, c, d = sorted_set
+    assert (a, b, c, d) == ('a', 'b', 'c', 'd')
+
+
+def test_unpacking_strings():
+    a, *tail = some_str
+    assert (a, tail) == ('a', ['b', 'c', 'd'])
+
+
+def test_unpacking_tuples():
+    a, *tail = some_tuple
+    assert (a, tail) == ('a', ['b', 'c', 'd'])
+
+
+def test_unpacking_lists():
+    a, *tail = some_list
+    assert (a, tail) == ('a', ['b', 'c', 'd'])
+
+
+def test_unpacking_sets():
+    a, *tail = sorted(some_set)
+    assert a == 'a'
+    assert set(tail) == {'b', 'c', 'd'}
+
+
+def test_unpacking_dicts():
+    a, *tail = some_dict
+    assert a == 'a'
+    assert set(tail) == {'b', 'c', 'd'}
+
+
+def test_unpacking_strings_head():
+    *head, d = some_str
+    assert (head, d) == (['a', 'b', 'c'], 'd')
+
+
+def test_unpacking_tuples_head():
+    *head, d = some_tuple
+    assert (head, d) == (['a', 'b', 'c'], 'd')
+
+
+def test_unpacking_lists_head():
+    *head, d = some_list
+    assert (head, d) == (['a', 'b', 'c'], 'd')
+
+
+def test_unpacking_sets_head_d():
+    *head, d = sorted(some_set)
+    assert set(head) == {'a', 'b', 'c'}
+    assert d == 'd'
+
+
+def test_unpacking_sets_head():
+    *head, d = sorted(some_set)
+    assert set(head) == {'a', 'b', 'c'} and d == 'd'
+
+
+def test_unpacking_dicts_head():
+    *head, d = some_dict
+    assert set(head) == {'a', 'b', 'c'}
+    assert d == 'd'
+
+
+def test_unpacking_strings_a_middle_d():
+    a, *middle, d = some_str
+    assert (a, middle, d) == ('a', ['b', 'c'], 'd')
+
+
+def test_unpacking_tuples_a_middle_d():
+    a, *middle, d = some_tuple
+    assert (a, middle, d) == ('a', ['b', 'c'], 'd')
+
+
+def test_unpacking_lists_a_middle_d():
+    a, *middle, d = some_list
+    assert (a, middle, d) == ('a', ['b', 'c'], 'd')
+
+
+def test_unpacking_sets_a_middle_d():
+    a, *middle, d = sorted(some_set)
+    assert a == 'a'
+    assert set(middle) == {'b', 'c'}
+    assert d == 'd'
+    assert a == 'a' and set(middle) == {'b', 'c'} and d == 'd'
+
+
+def test_unpacking_dicts_a_middle_d():
+    a, *middle, d = sorted(some_dict)
+    assert a == 'a'
+    assert set(middle) == {'b', 'c'}
+    assert d == 'd'
+    assert a == 'a' and set(middle) == {'b', 'c'} and d == 'd'
+
+
+def test_unpacking_strings_a_b_c_d_tail():
+    a, b, c, d, *tail = some_str
+    assert (a, b, c, d, tail) == ('a', 'b', 'c', 'd', [])
+
+
+def test_unpacking_tuples_a_b_c_d_tail():
+    a, b, c, d, *tail = some_tuple
+    assert (a, b, c, d, tail) == ('a', 'b', 'c', 'd', [])
+
+
+def test_unpacking_lists_a_b_c_d_tail():
+    a, b, c, d, *tail = some_list
+    assert (a, b, c, d, tail) == ('a', 'b', 'c', 'd', [])
+
+
+def test_unpacking_sets_a_b_c_d_tail():
+    a, b, c, d, *tail = sorted(some_set)
+    assert a == 'a'
+    assert b == 'b'
+    assert c == 'c'
+    assert d == 'd'
+    assert set(tail) == set()
```

Comment:
do not really understand why you convert to set before assertion

Reply:
fixed